src/client/datascience/notebook/notebookEditor.ts

```diff
@@ -256,6 +264,52 @@ export class NotebookEditor implements INotebookEditor {
         this._closed.fire(this);
     }
 
+    public runAbove(uri: Uri): void {
+        const cellId = this.getSelectedCellId(uri);
+        const index = this.document.cells.findIndex((c) => c.uri.toString() === cellId);
+
+        if (index > 0) {
+            // Get all cellIds until `index`.
+            const cells = this.document.cells.slice(0, index).map((cell) => cell);
+            this.runCellRange(cells);
+        }
+    }
+    public runCellAndBelow(uri: Uri): void {
+        const cellId = this.getSelectedCellId(uri);
+        const index = this.document.cells.findIndex((c) => c.uri.toString() === cellId);
+
+        if (index >= 0) {
+            // Get all cellIds starting from `index`.
+            const cells = this.document.cells.slice(index).map((cell) => cell);
+            this.runCellRange(cells);
+        }
+    }
+
+    private getSelectedCellId(uri: Uri): string {
+        const editor = this.vscodeNotebook.notebookEditors.find((nb) => nb.document.uri.toString() === uri.toString());
+
+        if (editor && editor.selection) {
+            return editor.selection.uri.toString();
+        }
+
+        return '';
```

Comment:
I'd make this function return string or undefined. While '' can work as a "not found value" the undefined make it more clear that you have something valid any string is returned and something not valid if undefined. Even if both empty string and undefined are falsy.